### ISSUE TITLE
Auto detect fails with braille displays.

### ISSUE THREAD

**Issue by stickbear2015** [nvaccess/nvda#9744]
### Steps to reproduce:
Connect a braille display via USB or bluetooth.
Set NVDA to auto detect the display in settings/braille.
### Actual behavior:
The display isn't re-detected after power cycling or re-connecting to USB from bluetooth or vice versa. It requires NVDA to be restarted to re-aquire braille.
### Expected behavior:
The display should be auto detected after powering off and back on and NVDA should not be required to restart to get braille back.
#### NVDA installed/portable/running from source:
Installed
#### NVDA version:
NVDA 2019.2.1 beta
#### Windows version:
Version 1903 (OS Build 18362.116)
#### Other information about your system:
Braille displays tested:
Focus 40 4th gen, via USB and blue tooth. firmward 5.71
Refreshabraille 18 running hid protocols with firmware 6.00 via USB.
Braille note Apex, USB and bluetooth
#### Does the issue still occur after restarting your PC?
yes.
#### Have you tried any other versions of NVDA? If so, please report their behaviors.
yes, nvda 2019.1.1 (the last stable release) same issues persist.

**Comment by stickbear2015** [comment]
The version number of the focus firmware is actually 5.71. my mistake.

**Comment by josephsl** [comment]
CC @LeonardDer

**Comment by DrSooom** [comment]
@stickbear2015: What happens if you restore the NVDA configuration by pressing NVDA+CTRL+R once? And please also read chapter 14.1 "Displays supporting automatic detection in the background" in NVDA's User Guide. Not all supported braille displays support the auto-detection feature yet.

**Comment by leonardder** [comment]
What do you mean with power cycling here?

If your display is connected via usb, it will only switch to bluetooth if:

* Bluetooth is on on the system and the device
* The device is paired
* USB is disconnected.

If your display is connected via Bluetooth, in NVDA 2019.1.1 it will only switch to usb if you disconnect bluetooth. In NVDA 2019.2, it should automatically switch to USB if you establish a USB connection.

**Comment by bramd** [comment]
@stickbear2015 I see that one of your displays is the Focus 40. Please retest with latest alpha or 2019.2 beta since the driver has been completely rewritten. As part of this rewrite, the display supports auto detection and the ability to automatically switch to USB if you plug in a USB cable and had a Bluetooth connection has been added. This ability was especially important for the Focus, to mirror the behavior of the old driver.

**Comment by stickbear2015** [comment]
I'm just getting back to this after a few days on business.
Can someone point me to the snapshots page for latest alphas, etc? As I seemed to have lost the link.

**Comment by DrSooom** [comment]
@stickbear2015: Here the [link to the NVDA Snapshots](https://www.nvaccess.org/files/nvda/snapshots/).

**Comment by stickbear2015** [comment]
Hello,
I've been testing this with the Braille Sense 6, and the original issue still persists.
I've also ran into an issue where if I restart the computer, and NVDA starts, and the braille display was already connected and working, the display doesn't come up and I have to disconnect the display, reconnect the display, and restart NVDA to make it play ball.
I do note that this happens with auto detect and specifically setting the display to the HIMS Braille Sense/Braille EDGE/Smart Beetle/Sync Braille series driver.
I could also be losing my mind, and doing something incorrectly.

**Comment by leonardder** [comment]
Could you please try this with NVDA 2023.3 Beta 1? Note that in that beta, in the braille display selection dialog, you can disable all drivers for auto detect you're not interested in.

**Comment by Adriani90** [comment]
@stickbear2015 can you please test with NVDA 2024.2?
As suggested by leonard above, please try to exclude the braille display drivers which are not related to your Braille display, in the list of drivers in NVDA Braille settings category.

**Comment by stickbear2015** [comment]
Firstly, I forgot this issue was open, lol.
So I now only have a Mantus q40 at my disposal, and I'd think, that I could use the Baum/HumanWare/APH/Orbit braille displays option, but in order to get the mantus display to detect I have to use the HumanWare Brailliant BI/B series / BrailleNote Touch option. Not even the standard HID display option works to detect this display. But I do not, using this option, see the original issues I reported in this issue happening anymore. So unless we want to troubleshoot why autodetect is being squirly and I'm having to use the HumanWare Brailliant BI/B series / BrailleNote Touch option instead of the Baum/HumanWare/APH/Orbit braille displays option, we can close this issue.

**Comment by seanbudd** [comment]
Thanks for getting back to us
